sklearn/utils/estimator_checks.py

```diff
@@ -290,6 +298,9 @@ def check_pipeline_consistency(name, Estimator):
                       random_state=0, n_features=2, cluster_std=0.1)
     X -= X.min()
     y = multioutput_estimator_convert_y_2d(name, y)
+    if name == "SpectralEmbedding":
+        # This data makes sure that the graph is fully connected.
+        X, y = S_X, S_y
```

Comment:
Okay! So I've (or I think I've) narrowed down the issue to a nearly singular graph laplacian when the graph is not fully connected...

* arpack's eigen solver seems to produce the non-deterministic eigen vectors when the laplacian matrix has a very low determinant... (order e-50) ( only "lobpcg" handles this perfectly producing deterministic output everytime... "amg" is not quite stable with nearly sigular matrices ( raises `LinAlgError` randomly ).)

* When this particular S_X, S_y is used the determinant of the laplacian is of the order e-20 which seems to be handled perfectly by `"arpack"`, `"lobpcg"` as well as `"amg"` `eigen_solvers`.

Could you kindly confirm if my observations are true?

We could fix this by reverting to `"lobpcg"` when the determinant of laplacian is very low (but involves an additional determinant computation step).

(Also note that we currently fall back to `"lobpcg"` solver when arpack raises `LigAlgError`...)